yt_dlp/extractor/maariv.py

```diff
@@ -0,0 +1,70 @@
+from .common import InfoExtractor
+import re
+
+
+class MaarivIE(InfoExtractor):
+    IE_NAME = 'maariv.co.il'
+    _VALID_URL = r'(https?://)?(www\.)?(m\.)?maariv\.co\.il(?:/[^/]+)*/Article-(?P<id>\d{7})'
+    _TESTS = [
+        {
+            'url': 'https://www.maariv.co.il/news/law/Article-1044008',
+            'info_dict': {
+                'id': '1044008',
+                'title': '.*',
+            },
+        }
+    ]
+
+    @staticmethod
+    def extract_resolution(url):
+        match = re.search(r'(\d{2,4}x\d{2,4})\.mp4$', url)
+        return match.group(1) if match else None
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        # Find the correct iframes
+        video_urls = re.findall(r'<iframe [^>]?poster="([^"]+)"[^>]+ src="([^"]+)"', webpage)
+        video_info_list = []
+
+        for thumbnail, src in video_urls:
+            media_param = re.search(r'media=(\d+)', src).group(1)
+            info_json = self._download_json(f"https://dal.walla.co.il/media/{media_param}?origin=player.maariv.co.il",
+                                            video_id)
+            data = info_json['data']
```

Comment:
Since there is the option of multiple videos being present, this should be non fatal:
```suggestion
            media_param = traverse_obj(parse_qs(video_obj.get('embedUrl')), ('media', 0))
            if not media_param:
                continue
            data = traverse_obj(self._download_json(
                f'https://dal.walla.co.il/media/{media_param}?origin=player.maariv.co.il',
                video_id, fatal=False), ('data', {dict}))
            if not data:
                continue
```